lib/http.c

```diff
@@ -2396,8 +2396,7 @@ CURLcode Curl_http_bodysend(struct Curl_easy *data, struct connectdata *conn,
        we don't upload data chunked, as RFC2616 forbids us to set both
        kinds of headers (Transfer-Encoding: chunked and Content-Length) */
     if(http->postsize != -1 && !data->req.upload_chunky &&
-       (conn->bits.authneg ||
-        !Curl_checkheaders(data, STRCONST("Content-Length")))) {
+       (!Curl_checkheaders(data, STRCONST("Content-Length")))) {
```

Comment:
Please explain why removing `conn->bits.authneg` from here is the correct move

Reply:
conn->bits.authneg is always false here, because in the block above, with conn->bits.authneg == true, we do not get here

Reply:
Ah, yes of course. Thanks!